app/models/king.rb

```diff
@@ -1,2 +1,8 @@
 class King < Piece
+
+  def valid_move?(new_x, new_y)
+      x_distance = (x_position - new_x).abs
+      y_distance = (y_position - new_y).abs    
```

Comment:
As you probably saw this did fail your tests. Try something like:
`(x_position - new_x).abs == 1 && (y_position - new_y).abs == 0
(x_position - new_x).abs == 0 && (y_position - new_y).abs == 1`
You might have to add some if elsif elsif else statments to make it all work.

Reply:
Ya I ran my tests and saw it failed but I needed to step away and start fresh today. I was thinking some if elsif statements, I'm gonna try that later.

Reply:
I am around today just send me a message on slack if you hit a block.